sky/provision/kubernetes/config.py

```diff
@@ -239,7 +256,9 @@ def _configure_autoscaler_service_account(
     if len(accounts) > 0:
         assert len(accounts) == 1
         logger.info('_configure_autoscaler_service_account: '
-                    f'{using_existing_msg(account_field, name)}')
+                    f'{updating_existing_msg(account_field, name)}')
+        kubernetes.core_api().patch_namespaced_service_account(
+            name, namespace, account)
```

Comment:
Is it possible to check the config of the account first and only patch the service account if it does not have all the config?
Reason: previously, we only required read permission for the user's account, but now we require write permission as well. This is to make sure the following works: the admin create the service account for all users and create restrictive user accounts with only read permission (this is what we made sure for the other clouds). Wdyt?

Reply:
Good point! I've updated checks to patch _only_ if the existing role does not match the required permissions.